src/main/scala/tech/cryptonomic/conseil/routes/QueryProtocol.scala

```diff
@@ -0,0 +1,36 @@
+package tech.cryptonomic.conseil.routes
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpjackson.JacksonSupport
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryOperations
+import tech.cryptonomic.conseil.tezos.QueryProtocolTypes.JsonQuery
+import tech.cryptonomic.conseil.util.RouteHandling
+
+import scala.concurrent.ExecutionContext
+
+/** Companion object providing apply implementation */
+object QueryProtocol {
+  def apply(implicit ec: ExecutionContext): QueryProtocol = new QueryProtocol()
+}
+
+/**
+  * Platform discovery routes.
+  * @param apiExecutionContext is used to call the async operations exposed by the api service
+  */
+class QueryProtocol(implicit apiExecutionContext: ExecutionContext) extends LazyLogging with RouteHandling with JacksonSupport {
+
+  val route: Route =
+    get {
+      entity(as[JsonQuery]) { query: JsonQuery =>
```

Comment:
I'll make unit tests for this route, but can you tell me what error would you expect it to return? now it returns 500, which is not descriptive, but I can change it

Reply:
so as I've investigated - we don't have any error handling on routes, and jackson does not want to return any viable error to handle on unsuccessful unmarshalling attempt(`NullPointerException`), I'll dig deeper with this - asked a question on gitter

Reply:
about the status: `400 BAD REQUEST` should do the job

Reply:
ok, added validation - because it looks like jackson uses runtime reflection it had to be done by checking fields(otherwise we would get NPE in the runtime), added tests to check for bad request response from endpoint